utils/model_validation_cache.py

```
@@ -0,0 +1,479 @@
+        """Remove expired entries from specified cache."""
+        expired_keys = []
+
+        for key, entry in cache_dict.items():
+            if self._is_expired(entry):
+                expired_keys.append(key)
+
+        for key in expired_keys:
+            del cache_dict[key]
+            logger.debug(f"Removed expired cache entry: {key}")
+
+    def cache_model_availability(
```

Pylint (3.3.7)

[refactor] Too many arguments (7/5)

(R0913)